hummingbot/connector/exchange/huobi/huobi_exchange.py

```diff
@@ -0,0 +1,423 @@
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderUpdate, TradeUpdate
+from hummingbot.core.data_type.trade_fee import TokenAmount, TradeFeeBase
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.event.events import MarketEvent
+from hummingbot.core.utils.async_call_scheduler import AsyncCallScheduler
+from hummingbot.core.utils.async_utils import safe_ensure_future
+from hummingbot.core.utils.estimate_fee import build_trade_fee
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+
+hm_logger = None
+s_decimal_0 = Decimal(0)
+s_decimal_NaN = Decimal("NaN")
+HUOBI_ROOT_API = "https://api.huobi.pro/v1"
+
+
+class HuobiExchange(ExchangePyBase):
+
+    UPDATE_ORDER_STATUS_MIN_INTERVAL = 1.0
+    web_utils = web_utils
+
+    MARKET_BUY_ORDER_COMPLETED_EVENT_TAG = MarketEvent.BuyOrderCompleted.value
+    MARKET_SELL_ORDER_COMPLETED_EVENT_TAG = MarketEvent.SellOrderCompleted.value
+    MARKET_ORDER_CANCELED_EVENT_TAG = MarketEvent.OrderCancelled.value
+    MARKET_TRANSACTION_FAILURE_EVENT_TAG = MarketEvent.TransactionFailure.value
+    MARKET_ORDER_FAILURE_EVENT_TAG = MarketEvent.OrderFailure.value
+    MARKET_ORDER_FILLED_EVENT_TAG = MarketEvent.OrderFilled.value
+    MARKET_BUY_ORDER_CREATED_EVENT_TAG = MarketEvent.BuyOrderCreated.value
+    MARKET_SELL_ORDER_CREATED_EVENT_TAG = MarketEvent.SellOrderCreated.value
+    API_CALL_TIMEOUT = 10.0
+    UPDATE_ORDERS_INTERVAL = 10.0
+    SHORT_POLL_INTERVAL = 5.0
+    LONG_POLL_INTERVAL = 120.0
+
+    def __init__(self,
+                 huobi_api_key: str,
+                 huobi_secret_key: str,
+                 trading_pairs: Optional[List[str]] = None,
+                 trading_required: bool = True):
+        self.huobi_api_key = huobi_api_key
+        self. huobi_secret_key = huobi_secret_key
+        self._trading_pairs = trading_pairs
+        self._trading_required = trading_required
+        self._account_id = ""
+        self._async_scheduler = AsyncCallScheduler(call_interval=0.5)
+        self._ev_loop = asyncio.get_event_loop()
+        super().__init__()
+        self._set_order_book_tracker(HuobiOrderBookTracker(connector=self,
+                                                           trading_pairs=trading_pairs,
+                                                           api_factory=self._web_assistants_factory,
+                                                           ))
+
+    @property
+    def name(self) -> str:
+        return "huobi"
+
+    @property
+    def authenticator(self):
+        return HuobiAuth(api_key=self.huobi_api_key, secret_key=self.huobi_secret_key)
+
+    @property
+    def rate_limits_rules(self):
+        return CONSTANTS.RATE_LIMITS
+
+    @property
+    def domain(self):
+        return
+
+    @property
+    def client_order_id_max_length(self):
+        pass
```

Comment:
According to the documentation the client-order-id max length is 64 characters. Please complete the method correctly and add the value as a global (https://huobiapi.github.io/docs/spot/v1/en/#place-a-new-order)